### PR TITLE
Feature/third web

### PR DESCRIPTION
**Pull request by mmahdigh**

### CODE REVIEW

**Review by mateodaza — APPROVED** [review]

File: src/hooks/useAuth.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    //   return false
+    // }
+  }
+
+  const estimateGasFee = async (
+    toAddress: string,
+    amount: string,
+    tokenAddress = ''
+  ) => {
+    // TODO: Rewrite the below logic with the third web methods
```
Comment (mmahdigh) [inline src/hooks/useAuth.ts:98]: Thirdweb has a built-in `estimateGas` method on its `Account` object (returned by `useActiveAccount` hook) that you can probably use. ([reference](https://portal.thirdweb.com/references/typescript/v5/Account#estimateGas))

Comment (mmahdigh) [reply src/hooks/useAuth.ts:98]: @mateodaza

Comment (mateodaza) [reply src/hooks/useAuth.ts:98]: Yeah. I will refactor all my work on another branch, we can comment for now @mmahdigh

Comment (mmahdigh) [reply src/hooks/useAuth.ts:98]: Sounds great!

File: src/lib/third-web/AutoConnect.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  const { connect } = useConnect()
+
+  useEffect(() => {
+    const main = async () => {
+      setIsAutoConnecting(true)
+      try {
+        const personalWalletId = localStorage.getItem(
+          LAST_CONNECT_PERSONAL_WALLET_ID
+        )
+        if (!personalWalletId) return
```
Comment (jainkrati) [inline src/lib/third-web/AutoConnect.tsx:28]: Error handling

File: src/lib/third-web/methods.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { inAppWallet, smartWallet } from "thirdweb/wallets";
+import { LAST_CONNECT_PERSONAL_WALLET_ID } from "./constants";
+import { client, smartWalletConfig } from "./provider";
+
+export const createSmartWallet = async () => {
+  const socialEOA = inAppWallet();
+  await socialEOA.connect({
+    client,
+    strategy: "google",
```
Comment (jainkrati) [inline src/lib/third-web/methods.ts:9]: will need to keep it dynamic. Lets take that in a separate PR.

**Review by jainkrati — APPROVED** [review]

**Pull request merged**